src/components/src/PurchaseOrderWidgetModal/purchase.order.widget.modal.tsx

```diff
@@ -0,0 +1,170 @@
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  generatePOPaymentPostBody() {
+    const { inputTextValue } = this.state;
+
+    return {
+      'limit-amount': '0',
+      'payment-instrument-identification-form': {
+        'display-name': 'aDisplayName',
+        'purchase-order': inputTextValue,
+      },
+    };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  createErrorMessageBody(postResultJson) {
+    if (postResultJson) {
+      const body = {
+        type: postResultJson.type,
+        debugMessages: postResultJson['debug-message'],
+        id: postResultJson.id,
+      };
+      return body;
+    }
+
+    return {};
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async createPOPaymentInstrument() {
+    const { createPaymentInstrumentActionUri, handleCloseModal } = this.props;
+    const { inputTextValue } = this.state;
+
+    const postResult = await cortexFetch(
+      `${createPaymentInstrumentActionUri}/?followlocation=true`,
+      {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+        },
+        body: JSON.stringify(this.generatePOPaymentPostBody()),
+      },
+    );
+
+    const postResultJson = await postResult.json();
+
+    if (postResult.status === 200 || postResult.status === 201) {
+      handleCloseModal();
+    } else {
+      this.setState({
+        isLoading: false,
+        errorMessage: postResultJson.messages ? this.createErrorMessageBody(postResultJson.messages[0]) : {},
+      });
+    }
+  }
+
+  updateInputState(event) {
+    const { errorMessage } = this.state;
+    if (errorMessage) {
+      this.setState({ inputTextValue: event.target.value, errorMessage: {} });
+    }
+    this.setState({ inputTextValue: event.target.value });
+  }
+
+  handleCloseMessageContainer() {
+    this.setState({ errorMessage: {} });
+  }
+
+  render() {
+    const { openModal, handleCloseModal } = this.props;
+    const { isLoading, errorMessage } = this.state;
+    const { inputTextValue } = this.state;
+
+    return (
+      <Modal open={openModal} onClose={handleCloseModal}>
+        <div className="modal-lg po-modal-container">
+          <div className="po-modal-content">
+            <div className="modal-header">
+              <h2 className="modal-title">
+                Add Purchase Order
+              </h2>
+            </div>
+            <div className="modal-body">
+              <form id="po_modal_form">
+                <div className="purchase-order-widget-modal-input-container">
+                  <input value={inputTextValue} className="form-control" type="text" placeholder="Enter Purchase Order Number (PO)" onChange={this.updateInputState} />
+                  {!_.isEmpty(errorMessage)
+                  && <MessageContainer message={[errorMessage]} handleCloseMessageContainer={this.handleCloseMessageContainer} />
```

Comment:
You shouldn't need to include MessageContainer again. It's included surrounding the app and should surface all Cortex debugMessages. This also allows for specific localized message overrides.

Reply:
For this one, feel free to reach out to @rostyk-kanafotskyy, @petro97, or @yulia-dnistrian if you need advice on handling the messages 😄